api4/channel.go

```diff
@@ -540,17 +540,20 @@ func deleteChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Allow delete if user is the only member left in channel
-	if memberCount > 1 {
-		if channel.Type == model.CHANNEL_OPEN && !app.SessionHasPermissionToChannel(c.Session, channel.Id, model.PERMISSION_DELETE_PUBLIC_CHANNEL) {
-			c.SetPermissionError(model.PERMISSION_DELETE_PUBLIC_CHANNEL)
-			return
-		}
+	// Do not allow delete if there's only one member left in a private channel
```

Comment:
That doesn't make any sense to me. Let's assume that the `RestrictPrivateChannelDeletion` policy is set to `team` and I'm a `team_admin` but not a `system_admin`. This means that when the channel has 2 members I am allowed to delete it, but if I am the only member, I am not allowed to delete it any more.